hass_nabucasa/acme.py

```diff
@@ -181,7 +181,9 @@ def _load_account_key(self) -> None:
             self.path_account_key.write_bytes(pem)
             self.path_account_key.chmod(0o600)
 
-        self._account_jwk = jose.JWKRSA(key=jose.ComparableRSAKey(key))
+        self._account_jwk = jose.JWKRSA(
+            key=jose.ComparableRSAKey(cast(rsa.RSAPrivateKey, key))
```

Comment:
Maybe we should just add a type annotation to the `key` local variabel at the top, and not initialize it to `None`?

Reply:
One of the functions have the expected return type.
The other one have a union which includes the correct one, so we would still need the casting.

Reply:
Is the key optional, is that the union? Why do we initialize it to `None`?

Reply:
It's not optional, it can never be none as the if or else will set it, but during init none would be correct as it's not yet set.

Reply:
Right. Can we remove the initialization at the top and just have a typed local variabel there? We know it will be set to a value in one of the two conditions.

Reply:
We can remove it, but it does not solve the problem with needing to cast the value.
> Argument 1 to "ComparableRSAKey" has incompatible type "Union[DHPrivateKey, Ed25519PrivateKey, Ed448PrivateKey, RSAPrivateKey, DSAPrivateKey, EllipticCurvePrivateKey, X25519PrivateKey, X448PrivateKey]"; expected "Union[RSAPrivateKey, RSAPublicKey, EllipticCurvePrivateKey, EllipticCurvePublicKey]"

We know it is `RSAPrivateKey`, but the defined type includes more variants that are not supported.

Reply:
Alternative workaround without casting <https://github.com/NabuCasa/hass-nabucasa/pull/491/commits/23d98752374f6504755ba024ff6092569a0478c9>